erigon-lib/state/inverted_index.go

```diff
@@ -76,6 +76,8 @@ type InvertedIndex struct {
 	// underlying array is immutable - means it's ready for zero-copy use
 	_visible *iiVisible
 	logger   log.Logger
+
+	enableReadAhead atomic.Bool
```

Comment:
it feels too-much: 
- we have same atomics inside files, now we have them in ii/h/d, then likely will have in agg or higer level. 
- feels-too much... like over-engineering
- also atomic.Bool does "technically" protect from double decrement of underlying atomics. but it's not "solving" - it "hiding" problem:
```
// high-level code: like snap_cmd.go
a.MadvEnable()
     // low-level code: like build IDX
     file.MadvEnable()
     file.Disable()

// high-level code: working with disabled Madv (because low-level code disabled it)

a.Disable()
```